libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/internal/route/line/MapboxRouteLineUtils.kt

```diff
@@ -319,6 +319,49 @@ object MapboxRouteLineUtils {
         return routeLineTrafficData
     }
 
+    @SuppressLint("LogNotTimber")
+    fun getRestrictedRouteSections(route: DirectionsRoute): List<List<Point>> {
+        try {
+            val coordinates = LineString.fromPolyline(
```

Comment:
I wanted to discuss the possibility of using `RoadObjectObserver`. `RoadObjectObserver` gives access to `List<RoadObject>` where a `RoadObject` can be of several different types containing `RestrictedAreaEntrance` and `RestrictedAreaExit` with specific geometry. 
This observer is triggered, every time a new route is fetched. This object is also available in `RouteProgress` in the form of `routeProgress.upcomingRoadObjects`. 
I think hooking up to this observer will be beneficial in the long run with a more full proof implementation, because with this, the user can see blockades right away when a route is requested and also along the route progress. It also probably saves all the business logic you have here. 
Thoughts?
cc @cafesilencio @Guardiola31337 @LukasPaczos

Reply:
How is an observer going to stay synched up with the developer's calls to setRoutes?

Reply:
I like the idea of using the `RoadObjectObserver` because we don't duplicate the calculations that already happened in NN, but the problem is that then we rely on the route being set as the primary route. If someone would like to use the `RouteLineApi` to show a preview of the route, they wouldn't be able to show the restricted areas.

I think we need to continue as-is for now, and iterate if needed.